docs/pages/Cardinal/Cross_Shard_Communication/evm_to_cardinal.mdx

```diff
@@ -0,0 +1,188 @@
+
+
+### messageResult
+
+The `messageResult` method enables smart contracts to retrieve the result of a cross-shard message.
+
+#### Parameters
+
+| Parameter | Description                                                            |
+|-----------|------------------------------------------------------------------------|
+| `txHash`  | The hash of the EVM transaction that triggered the `sendMessage` call. |
+
+
+#### Return Values
+
+| Type     | Description                                        |
+|----------|----------------------------------------------------|
+| `bytes`  | ABI encoded result struct.                         |
+| `string` | Error string, if any. Empty string means no error. |
+| `uint32` | A numeric value representing the result status.    |
+
+#### Codes
+
+| Code                     | Value | Meaning                                            |
+|--------------------------|-------|----------------------------------------------------|
+| `Success`                | `0`   | Transaction executed successfully.                 |
+| `TxFailed`               | `1`   | Transaction execution failed.                      |
+| `NoResult`               | `2`   | No result available for the operation.             |
+| `ServerUnresponsive`     | `3`   | Game Shard is unresponsive.                        |
+| `Unauthorized`           | `4`   | Unauthorized access or action.                     |
+| `UnsupportedTransaction` | `5`   | Transaction type is not supported.                 |
+| `InvalidFormat`          | `6`   | Data or format is invalid.                         |
+| `ConnectionError`        | `100` | Error in establishing or maintaining a connection. |
+| `ServerError`            | `101` | Internal error with the game shard.                |
+
+### query
+
+The `query` router method enables smart contracts to read data from a game shard specified by the given namespace.
+
+#### Parameters
+
+| Parameter   | Type     | Description                                         |
+|-------------|----------|-----------------------------------------------------|
+| `request`   | `bytes`  | ABI encoded query request struct.                   |
+| `resource`  | `string` | The resource identifier for the query. (query name) |
+| `namespace` | `string` | The namespace of the game shard.                    |
+
+#### Return Values
+
+| Type    | Description                      |
+|---------|----------------------------------|
+| `bytes` | ABI encoded query result struct. |
+
+## Structuring Messages and Query Requests
+
+The `query` and `sendMessage` Router methods both take in a parameter of type `bytes`. These bytes must be formed by ABI encoding a struct with the exact same field types as their Cardinal game shard counterparts.
+
+For example:
+
+If the Cardinal game shard defines this message input:
+```go
+type SendEnergy struct {
+    PlanetTo string
+    PlanetFrom string
+    Amount uint64
+}
+```
+
```

Comment:
I think this section needs a brief description on what solidity is and links for users to follow to learn more. 

Similar to this tutorial:

https://sotrh.github.io/learn-wgpu/

The very beginning of the tutorial assumes users don't even know rust or know the rust package manager and it gives people links to learn more.

Reply:
added a callout to the top, lmk how that looks